src/drivers/Bma421.cpp

```diff
@@ -24,6 +24,17 @@ namespace {
   }
 }
 
+// Scale factors to convert accelerometer counts to milli-g
+// from datasheet: https://files.pine64.org/doc/datasheet/pinetime/BST-BMA421-FL000.pdf
+// The array index to use is stored in accel_conf.range
+const short Bma421::accelScaleFactors[] = {
+  1024,  // LSB/g +/- 2g range
+  512,   // LSB/g +/- 4g range 
+  256,   // LSB/g +/- 8g range
+  128    // LSB/g +/- 16g range
+};
```

Comment:
This should be `constexpr` rather than `const`, so it can be evaluated at compile-time and `int16_t` instead of `short`. I would also move it into the anonymous namespace above so that it's only available in the current compilation unit. I would also recommend that you explicitly set the indices.

```suggestion
constexpr int16_t accelScaleFactors[] = {
  [BMA4_ACCEL_RANGE_2G] = 1024,  // LSB/g +/- 2g range
  [BMA4_ACCEL_RANGE_4G] = 512,   // LSB/g +/- 4g range 
  [BMA4_ACCEL_RANGE_8G] = 256,   // LSB/g +/- 8g range
  [BMA4_ACCEL_RANGE_16G] = 128   // LSB/g +/- 16g range
};
```

Reply:
Setting the indices explicitly is very neat - I didn't know you could do that, thanks (I don't know any of the C++ fancy bits - I'm just trying to treat it like C with classes!).

Reply:
Setting the indices is even possible in plain old C as well! As long as you're using C99 or later that is.

Reply:
Ha!  C99 isn't 'plain old C' - I remember the debates about whether to use K&R or ANSI C coding style in a project!